rosidl_generator_cpp/resource/msg__struct.hpp.template

```diff
@@ -144,6 +152,14 @@ for field in spec.fields:
   typedef std::weak_ptr<@(cpp_full_name)<ContainerAllocator> const>
     ConstWeakPtr;
 
+  // pointer types similar to ROS 1, use SharedPtr / ConstSharedPtr instead
+  typedef @(deprecated_macro_name)
+    std::shared_ptr<@(cpp_full_name)<ContainerAllocator>>
+    Ptr;
+  typedef @(deprecated_macro_name)
+    std::shared_ptr<@(cpp_full_name)<ContainerAllocator> const>
+    ConstPtr;
```

Comment:
Putting it at the end doesn't work on Windows (http://ci.ros2.org/job/ros2_batch_ci_windows/146/console). On Linux / OS X it seems to work in both places.

Should I add it to the platform specific location (making the template and code a bit more complex) or just keep it where it is (http://ci.ros2.org/job/ros2_batch_ci_windows/147/console)?